Audio enable corrections (2/3) (#8903)
qmk/qmk_firmware: docs/ChangeLog/20200530/PR8903.md

## fixing wrong configuration of AUDIO feature

> * audio_avr.c does not default to any pin, there has to be a #define XX_AUDIO in config.h at some level, for it to actually work
>   otherwise the audio-code ends up cluttering the firmware, possibly breaking builds because the maximum allowed firmware size is exceeded
> 
> => these changes whant to fix this by disabling audio on keyboards that have audio misconfigured=non-functional

Reviewer: Some wordsmithing.

```suggestion
`audio_avr.c` does not default to any pin; there has to be a #define XX_AUDIO in config.h at some level for Audio to actually work. Otherwise, the Audio code ends up cluttering the firmware, possibly breaking builds because the maximum allowed firmware size is exceeded.

These changes fix this by disabling Audio on keyboards that have the feature misconfigured, and therefore non-functional.
```
